src/client/sandbox/event/message.js

```diff
@@ -172,12 +172,19 @@ export default class MessageSandbox extends SandboxBase {
         if (!this._isIframeRemoved() && (isIframeWithoutSrc || !isCrossDomainWindows(targetWindow, this.window) &&
                                                                targetWindow[this.RECEIVE_MSG_FN])) {
             //NOTE: postMessage delay imitation
-            nativeMethods.setTimeout.call(this.topWindow, () =>
+            nativeMethods.setTimeout.call(this.topWindow, () => {
+                // NOTE: We leave the capability to communicate with a removed frame. Unfortunately, this
+                // cannot be done in Safari, because it restricts operations with removed windows (GH-171).
+                var isIframeRemoved = !this.window.top;
```

Comment:
Moreover, `removed` iframe is already used here: https://github.com/churkin/testcafe-hammerhead/blob/i171/src/client/sandbox/event/message.js#L104

But it has quite different meaning. And still think we should get rid of Safari check. Seems like removed and reachable iframe will still have `window.top`